app/controllers/users_controller.rb

```diff
@@ -28,9 +28,12 @@ def edit
     @constituencies = OnsConstituency.all.order(:name)
   end
 
+  # rubocop:disable Metrics/MethodLength
   def update
+    review_required = false
     if params[:user]
       @user.assign_attributes(user_params)
+      review_required = @user.swap_profile_changed?
```

Comment:
Seems quite cunning - I guess this way you don't need a new dirty flag in the DB and just rely on `ActiveRecord::Dirty` instead.

Reply:
Exactly

Reply:
@baob There's a problem with not tracking "willing party has been reviewed and confirmed" in the db, which is that all the existing users without any swaps will never see this page because they've already set their willing.

So I think it's worth having a db field which tracks this to ensure everyone goes through the confirmation of their willing choice.